src/components/projects/wbs-details/work-package-container/work-package-container-view/work-package-edit-details/work-package-edit-details.tsx

```diff
@@ -106,33 +112,34 @@ const WorkPackageEditDetails: React.FC<Props> = ({ workPackage, users, setters }
     );
   };
 
-  const buildUsersSelect = (title: string, defaultUser: User | undefined, onChange: any) => {
-    if (!defaultUser) {
-      return (
-        <>
-          <b>{title}</b>
-          <Form.Control as="select" onChange={(e) => onChange(e.target.value)}>
-            {users.map((user, index) => (
-              <option key={user.userId} value={fullNamePipe(user)}>
-                {fullNamePipe(user)}
-              </option>
-            ))}
-          </Form.Control>
-          <br />
-        </>
-      );
+  const buildUsersSelect = (
+    title: string,
+    defaultUser: User | undefined,
+    updateUser: (val: number) => void
+  ) => {
+    let otherUsers = users;
+    if (defaultUser !== undefined) {
+      otherUsers = users.filter((user) => user.userId !== defaultUser.userId);
     }
-
-    const otherUsers = users.filter((user) => user.userId !== defaultUser.userId);
     return (
       <>
         <b>{title}</b>
-        <Form.Control as="select" onChange={(e) => onChange(e.target.value)}>
-          <option key={defaultUser.userId} value={fullNamePipe(defaultUser)}>
-            {fullNamePipe(defaultUser)}
-          </option>
-          {otherUsers.map((user, index) => (
-            <option key={user.userId} value={fullNamePipe(user)}>
+        <Form.Control
+          as="select"
+          data-testid={title}
+          onChange={(e) => updateUser(parseInt(e.target.value))}
+        >
+          {defaultUser === undefined ? (
+            <option key={-1} value={-1}>
+              {emDashPipe('')}
+            </option>
+          ) : (
+            <option key={defaultUser.userId} value={defaultUser.userId}>
```

Comment:
the value here has to be the same time of value it was before becuase Justin is using different states than I was